convert2rhel/actions/__init__.py

```diff
@@ -206,25 +211,122 @@ def run(self):
 
         self._has_run = True
 
-    def set_result(self, status=_NO_USER_VALUE, error_id=_NO_USER_VALUE, message=_NO_USER_VALUE):
+    @property
+    def result(self):
+        """
+        Helper method to save result variable in a different variable, so it does
+        not get overwritten by the __init__ method of the Action class
+        """
+        return self._result
+
+    @result.setter
+    def result(self, action_message):
+        """
+        Make sure the correct policy for result messages is used.
         """
-        Helper method that sets the resulting values for status, error_id and message.
+        if not isinstance(action_message, ActionResult):
+            raise Exception()
 
-        :param status: The status to be set.
-        :type: status: str
-        :param error_id: The error_id to identify the error.
-        :type error_id: str
+        self._result = action_message
+
+    def set_result(self, level, id=None, message=None):
+        """
+        Helper method that sets the resulting values for level, id and message.
+
+        :param id: The id to identify the result.
+        :type id: str
+        :param level: The status_code of the result .
+        :type: level: str
         :param message: The message to be set.
         :type message: str | None
         """
-        if status != _NO_USER_VALUE:
-            self.status = STATUS_CODE[status]
+        if level not in ("ERROR", "OVERRIDABLE", "SKIP", "SUCCESS"):
+            raise KeyError("The level of result must be FAILURE, OVERRIDABLE, SKIP, or SUCCESS.")
+
+        self.result = ActionResult(level, id, message)
+
+    def set_message(self, level, id=None, message=None):
```

Comment:
This shouldn't be named `set_` because it doesn't set the message attribute (`message = FOO`).

`add_` was used before which is similar to `set` but there could be other prefixes that work as well.

Reply:
add probably works best for now